src/weapon/w_045.c

```diff
@@ -41,7 +41,63 @@ INCLUDE_ASM("weapon/nonmatchings/w_045", EntityWeaponAttack);
 
 INCLUDE_ASM("weapon/nonmatchings/w_045", func_ptr_80170004);
 
-INCLUDE_ASM("weapon/nonmatchings/w_045", func_ptr_80170008);
+void func_ptr_80170008(Entity* self) {
+    SetSpriteBank1(D_13F000_8017A040);
+    if (g_HandId != 0) {
+        g_CurrentEntity->animSet = ANIMSET_OVL(0x12);
+        g_CurrentEntity->palette = 0x128;
+        g_CurrentEntity->unk5A = 0x66;
+    } else {
+        g_CurrentEntity->animSet = ANIMSET_OVL(0x10);
+        g_CurrentEntity->palette = 0x110;
+        g_CurrentEntity->unk5A = 0x64;
+    }
+    SetSpeedX(FIX(2.5));
+    if (!(g_Player.pl_vram_flag & 1)) {
+        PLAYER.step = 0x2A;
+        PLAYER.step_s = 0;
+        PLAYER.velocityY = 0;
+        PLAYER.ext.player.anim = 0xD0;
+        PLAYER.animFrameIdx = 0;
+        PLAYER.animFrameDuration = 0;
+        return;
+    }
+    if (func_13F000_8017A718() == 0) {
+        PLAYER.step = 0x28;
+        PLAYER.step_s = 0;
+        PLAYER.ext.player.anim = 0xCF;
+        PLAYER.animFrameIdx = 0;
+        PLAYER.animFrameDuration = 0;
+        PLAYER.velocityX >>= 1;
+        return;
+    }
+    if (g_Player.padTapped & PAD_SQUARE) {
+        PLAYER.ext.player.anim = 0xCE;
+        g_api_PlaySfx(0x6F0);
```

Comment:
I think we're trying to use enums for all SFX moving forward

Reply:
Should also be `g_api.PlaySfx`, will fix

Reply:
I'm tempted to hold off creating enums for these until https://github.com/Xeeynamo/sotn-decomp/pull/1441#issuecomment-2253470492 is resolved but I'll add these two

Reply:
Fair, I think in general I'm in favor of over-documenting, if we end up duplicating entries or whatever it's easier to consolidate them later than to hunt through the repo digging for undocumented uses